corehq/form_processor/backends/couch/processor.py

```diff
@@ -81,6 +82,27 @@ def hard_delete_case_and_forms(cls, domain, case, xforms):
         docs = [case._doc] + [f._doc for f in xforms]
         case.get_db().bulk_delete(docs)
 
+    @classmethod
+    def new_form_from_old(cls, xml, xform, value_responses_map, user_id):
+        from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+        from corehq.form_processor.parsers.form import apply_deprecation
+        existing_form = FormAccessors(xform.domain).get_with_attachments(xform.get_id)
```

Comment:
This line is common to couch and sql, could move it up to the `FormProcessorInterface` method.